colcon_core/task/python/build.py

```diff
@@ -190,8 +190,16 @@ def _symlinks_in_build(self, args, setup_py_data):
         if os.path.exists(os.path.join(args.path, 'setup.cfg')):
             items.append('setup.cfg')
         # add all first level packages
-        items += list(filter(
-            lambda name: '.' not in name, setup_py_data['packages']))
+        package_dir = setup_py_data.get('package_dir', {})
+        for package in setup_py_data['packages']:
```

Comment:
```suggestion
        for package in setup_py_data.get('packages', []):
```

Reply:
I first read your comment and didn't see the suggestion yet. I already fixed it in affcb3246ae722f689fb46ca0483be518ccb0987 in actually two locations.